### PR TITLE
Use Freedom Typescript Api

### PR DESCRIPTION
**Pull request by iislucas**
Hi Trevor, this is a followup on your CL to use the npm typescript API module. Does a little naming cleanup too.

### CODE REVIEW

File: Gruntfile.coffee [hunk 1]
```diff
@@ -38,7 +38,8 @@ module.exports = (grunt) ->
     # All typescript compiles to build/ initially.
     ts: {
       socks2rtc: {
-        src: ['src/interfaces/*.d.ts',
+        src: ['node_modules/freedom-typescript-pi/interfaces/*.d.ts',
```
Comment (trevj) [inline Gruntfile.coffee:41]: As discussed, looks like these aren't needed.

File: src/chrome-providers/chrome-tcpsocket.ts [hunk 2]
```diff
@@ -20,28 +25,38 @@ module Sockets {
    *   (http://developer.chrome.com/apps/socket.html)
    * for the freedom interface.
    */
-  export class Chrome implements Sockets.API {
+  export class Chrome implements TcpSocket.Implementation {
 
     constructor (
         private channel,
-        private dispatchEvent:(event:string,data:any)=>void) {
+        private dispatchEvent:(event:string,data:Object)=>void) {
```
Comment (trevj) [inline src/chrome-providers/chrome-tcpsocket.ts:32]: do this for chrome-udpsocket.ts too?

File: src/chrome-providers/chrome-tcpsocket.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+*
+* Implements: freedom-typescript-api/interfaces/tcp-socket.d.ts
+*
+* TODO: This should be refactored into freedom someday...
+*/
+/// <reference path='../../node_modules/freedom-typescript-api/interfaces/tcp-socket.d.ts' />
+/// <reference path='../../node_modules/freedom-typescript-api/interfaces/promise.d.ts' />
+
+var Sockets;
+(function (Sockets) {
```
Comment (trevj) [inline src/chrome-providers/chrome-tcpsocket.js:12]: I don't think this file should be here!

**Comment by trevj** [comment]
Overall I like it -- just a couple of little things then I think it's good to go!

**Comment by iislucas** [comment]
Thanks! Fixes in a followup CL coming very soon.

**Pull request merged**